src/wallet/asyncrpcoperation_sendmany.cpp

```diff
@@ -979,38 +1059,24 @@ UniValue AsyncRPCOperation_sendmany::perform_joinsplit(
             outputMap,
             info.vpub_old,
             info.vpub_new,
-            !this->testmode,
+            generateProof,
             &esk); // parameter expects pointer to esk, so pass in address
-    {
+
+    if (generateProof) {
         auto verifier = libzcash::ProofVerifier::Strict();
```

Comment:
I'm confused how this code worked before without checking `!this-<testmode`

Reply:
Thanks for raising this! I have looked into it, and it turns out that the test was expecting this function to throw an error (and looking for it), but it didn't enforce the error - it only checked the returned error message, if any. That's why I didn't spot the effect of this change, because tests kept passing. But while investigating, I found a separate bug that was introduced later in this PR, that *did* cause the test to fail! I'll fix both of these :smiley:

Reply:
I guess you mean this line   https://github.com/zcash/zcash/pull/2120/files#diff-0745db44791e51061a62e0bb159c102b
Seems additional enforcing needs to be added here: 
https://github.com/zcash/zcash/blob/master/src/test/rpc_wallet_tests.cpp#L1372

Reply:
Was that fixed?